src/main/java/picard/illumina/BasecallsConverterBuilder.java

```diff
@@ -0,0 +1,119 @@
+package picard.illumina;
+
+import htsjdk.samtools.SAMRecordQueryNameComparator;
+import htsjdk.samtools.util.SortingCollection;
+import picard.illumina.parser.ReadStructure;
+import picard.illumina.parser.readers.BclQualityEvaluationStrategy;
+
+import java.io.File;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Map;
+
+public class BasecallsConverterBuilder<CLUSTER_OUTPUT_RECORD> {
+
+    File basecallsDir;
+    File barcodesDir;
+    int lane;
+    ReadStructure readStructure;
+
+    Comparator<CLUSTER_OUTPUT_RECORD> outputRecordComparator;
+    SortingCollection.Codec<CLUSTER_OUTPUT_RECORD> codecPrototype;
+    Class<CLUSTER_OUTPUT_RECORD> outputRecordClass;
+
+    Map<String, ? extends BasecallsConverter.ConvertedClusterDataWriter<CLUSTER_OUTPUT_RECORD>> barcodeRecordWriterMap;
+    boolean demultiplex = false;
+    int maxReadsInRamPerTile = 1200000;
+    List<File> tmpDirs = Collections.singletonList(new File(System.getProperty("java.io.tmpdir")));
```

Comment:
Is there not a more picard-way to get a tmp dir?  Something that would respect `TMP_DIR=` from the command line?